### ISSUE TITLE
Add voting on comments

### ISSUE THREAD

**Issue by Quix0r** [Chocobozzz/PeerTube#1644]
Like with videos, also comments should be voted (up/down) on, increasing quality of comments.

**Comment by Zig-03** [comment]
I personally think it's a fine feature as long the numbers get hidden by default, following the same idea mentioned here https://github.com/Chocobozzz/PeerTube/issues/1302

**Comment by Quix0r** [comment]
Default should be set on a per-channel basis?

**Comment by XenonFiber** [comment]
I think that any rating system already implemented or to be implemented should be able to be disabled by the user or completely disabled by the admin.

**Comment by buoyantair** [comment]
I do think it's a good thing to have a rating system. But please make it so that admins cannot control the rating system (unless they themselves dig into the code and modify the server). The rating system must all costs be working under a democratic ideal. Although, if server admins want to, they can always hide the number of votes on all comments.

**Comment by Quix0r** [comment]
Ratings are good as they increase quality of posts/comments. Please see stackoverflow for an excellent example of ratings on comments (only up or report). Ratings < 0 should be faded out (opacity) to draw away attention from it.

**Comment by Quix0r** [comment]
I agree with that the admin has the "last word" of entirely disabling rating, then the uploader (user) can disable it in profile (all videos) or per video.

**Comment by buoyantair** [comment]
@Zig-03 I don't think ratings on comments should be hidden. It's one thing to have a particular video's ratings be hidden, as you said, to avoid peer pressure & unwanted emotional trauma. It's completely different thing when you want to have a serious discussion & upvote decent opinions. The comment system should be a place for reasoning & voting helps boast the most relatable reason. A good example is Hackernews.

**Comment by Booteille** [comment]
I tend to agree with @Zig-03 that votes should be hidden.
I think we don't need to know how much a commentary received endorsements even if we need voting to have a soft moderation on comments.
Upvoted comments should have a better visibility where too much downvoted comments should be hidden by default. With the better visibility on useful comments, I think showing how much votes a comment have is less interesting and will influence how users will vote.

**Comment by Quix0r** [comment]
Entirely hidden or only visible to admin/logged-in user and hidden to guests? Maybe consider implementing something like stackoverflow has? Similar as @buoyantair says, it encourages people to write good-quality comments and not nonsense. But if the admin finds it okay, let him disable/hide the rating system.

**Comment by trymeouteh** [comment]
Would like to see a upvote and downvotes on comments

**Comment by 3risian** [comment]
I feel that a lot of these preferences for vote visibly and comment hiding threshold are very context dependent, so admin and/or channel control of these things are very important.

Simple voting with sorting and vote visibility are very effective in contexts where replies are reasonably objective, like Stack Overflow. Tech and education instances, for instance, could use this to their advantage.
The same sorting can be destructive in more subjective contexts where differing opinions can be just as valid. In these contexts, diversity of opinion deserves more respect instead of simply lowering visibility of controversial comments. Censoring minority opinions tends to radicalize communities.

**Comment by kontrollanten** [comment]
Here's a rough proposal for how it can look like. Comments are sorted by most recent and down voted comments get faded. In the long run I guess we'd like to sort the comments based on the number of votes.

![image](https://user-images.githubusercontent.com/6680299/113268084-4fd64b00-92d7-11eb-9ad0-a32c136c56fc.png)

**Comment by ar9708** [comment]
I think @3risian raise a valid point, and depending on the site different combinations of comment-likes should be configurable. Could we consider adding these with feature flags, hidden and working as now by default? E.g.

* `commentLikeButtonEnabled=yes/no`
* `commentDislikeButtonEnabled=yes/no`
* `commentLikeCountVisible=yes/no`
* `commentDislikeCountVisible=yes/no`
